packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -298,6 +301,148 @@ class RenderParagraph extends RenderBox
     markNeedsLayout();
   }
 
+
+  /// The selection service to enable text selection, or `null` if text selection
+  /// is disabled.
+  SelectionRegistrar? get selectionRegisrar => _selectionRegistrar;
+  SelectionRegistrar? _selectionRegistrar;
+  set selectionRegisrar(SelectionRegistrar? value) {
+    if (value == selectionRegisrar)
+      return;
+    selectionRegisrar?.remove(this);
+    _selectionRegistrar = value;
+    selectionRegisrar?.add(this);
+  }
+
+  TextSelection? _textSelection;
+
+  @override
+  String? copy() {
+    final TextSelection? textSelection = _textSelection;
+    if (textSelection == null)
+      return null;
+
+    final String plainText = text.toPlainText(includeSemanticsLabels: false);
+    return plainText.substring(textSelection.start, textSelection.end);
+  }
+
+  /// Clear the current text selection, but only mark for a paint if it has
+  /// been set to a non-null value.
+  @override
+  void clearSelection() {
+    _dragSelectionStartInGlobalCoordinate = null;
+    if (_textSelection == null)
+      return;
+    _textSelection = null;
+    markNeedsPaint();
+  }
+
+  @override
+  Rect get globalRect {
+    if (!attached) {
+      return Rect.zero;
+    }
+    final Rect boundingRect = Rect.fromLTWH(0, 0, _textPainter.size.width, _textPainter.size.height);
+    final Matrix4 transform = getTransformTo(null);
+    return MatrixUtils.transformRect(transform, boundingRect);
+  }
+
+  @override
+  void selectAll() {
+    final TextSelection textSelection = TextSelection(
+      baseOffset: 0,
+      extentOffset: text.toPlainText(includeSemanticsLabels: false).length
+    );
+    if (textSelection != _textSelection) {
+      _textSelection = textSelection;
+      markNeedsPaint();
+    }
+  }
+
+  Offset? _dragSelectionStartInGlobalCoordinate;
+
+  @override
+  SelectionResult dispatchSelectionEvent(SelectionEvent event) {
+    if (!attached)
+      return SelectionResult.none;
+    final TextSelection? existingSelection = _textSelection;
+    late final SelectionResult result;
+    if (event is DragSelectionEvent) {
+      if (event is DragSelectionStartEvent) {
+        _dragSelectionStartInGlobalCoordinate = event.offset;
```

Comment:
Wouldn't we want to hold on to the start point in local coordinates und use that for the calculations in _updateDragSelection? 

I am thinking about a scenario where you start the selection with a pointer down in a large body of text and while the pointer is down you scroll a surrounding scrollable. I think that scroll needs to extend the selection and the selection needs to start at the later that the pointer touched when it cam down first. Due to the scroll, that letter will have moved in the global coordinate system, though.